firefox-ios/firefox-ios-tests/Tests/ClientTests/TrackingProtectionTests/CertificatesModelTests.swift

```diff
@@ -0,0 +1,47 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import XCTest
+import X509
+@testable import Client
+
+final class CertificatesModelTests: XCTestCase {
+    private var viewModel: CertificatesViewModel!
+
+    override func setUp() {
+        super.setUp()
+        viewModel = CertificatesViewModel(topLevelDomain: "topLevelDomainTest.com",
+                                          title: "TitleTest",
+                                          URL: "https://google.com",
+                                          certificates: [])
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        viewModel = nil
+    }
+
+    func testGetCertificateValues() {
+        let data = "CN=www.google.com, O=Google Trust Services, C=US"
+        let result = viewModel.getCertificateValues(from: data)
+        XCTAssertEqual(result["CN"], "www.google.com")
+        XCTAssertEqual(result["O"], "Google Trust Services")
+        XCTAssertEqual(result["C"], "US")
+    }
+
+    func testGetCertificateValuesWithMissingValue() {
+        let data = "CN=www.google.com, O=, C=US"
+        let result = viewModel.getCertificateValues(from: data)
+        XCTAssertEqual(result["CN"], "www.google.com")
+        XCTAssertEqual(result["O"], "")
+        XCTAssertEqual(result["C"], "US")
+    }
+
+    func testGetDNSNamesList() {
+        let input = #"DNSName("www.google.com"), DNSName("*www.google.com")"#
+        let result = viewModel.getDNSNamesList(from: input)
+        XCTAssertEqual(result, ["www.google.com", "*www.google.com"])
+    }
```

Comment:
Should we maybe add tests where you pass an empty string as input to your new functions?

e.g. `let result = viewModel.getCertificateValues(from: "")`
e.g. `let result = viewModel.getDNSNamesList(from:"")`

I would like to know how your code handles bad input.

You might also want to think about what happens if an empty `[]` or `[:]` is returned in CertificatesViewController, for example. (I'll leave a comment in `setupCertificatesHeaderView`.)